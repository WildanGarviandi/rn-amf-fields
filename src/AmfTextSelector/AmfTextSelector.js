import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { Dimensions, ScrollView, StyleSheet, Text, TouchableOpacity, View } from 'react-native'
import { interpolateRdYlGn } from 'd3-scale-chromatic'

let { width, height } = Dimensions.get('window')

class AmfTextSelector extends Component {
  constructor(props) {
    super(props)
  }

  renderItems(item, index) {
    let color
    if (item.color == undefined) {
      let interpolation = 0
      if (index < this.props.items.length/2) interpolation = index/this.props.items.length
      else interpolation = (index+1)/this.props.items.length
      color = interpolateRdYlGn(interpolation)
    } else color = item.color

    let selected = this.props.selectedIndex == index

    return (
      <TouchableOpacity key={index} style={style.container}
        onPress={() => {
          this.props.onPress(index)
        }}
      >
        <View style={[style.indicator, {backgroundColor: selected ? '#421C58' : color}]}>
        </View>
        <View style={[style.textContainer, {backgroundColor: selected ? '#421C58' : 'white', borderColor: selected ? '#421C58' : '#D4D4D4'}]}>
          <Text style={{color: selected ? 'white' : 'black'}}>{item.label}</Text>
        </View>
      </TouchableOpacity>
    )
  }

  render() {
    return (
      <ScrollView
        horizontal={true}
        style={this.props.style}
        contentContainerStyle={{padding: 16}}
      >
        { this.props.items.map(this.renderItems.bind(this)) }
      </ScrollView>
    )
  }
}

const style = StyleSheet.create({
  container: {
    marginHorizontal: 8,
    width: width*0.6
  },
  indicator: {
    height: 10,
    borderTopLeftRadius: 4,
    borderTopRightRadius: 4
  },
  textContainer: {
    padding: 16,
    borderRightWidth: 1,
    borderLeftWidth: 1,
    borderBottomWidth: 1
  }
})

const itemShape = PropTypes.shape({
  color: PropTypes.string,
  label: PropTypes.string.isRequired,
  value: PropTypes.any.isRequired
})

AmfTextSelector.propTypes = {
  items: PropTypes.arrayOf(itemShape).isRequired,
  onPress: PropTypes.func.isRequired,
  selectedIndex: PropTypes.number.isRequired
}

export default AmfTextSelector
