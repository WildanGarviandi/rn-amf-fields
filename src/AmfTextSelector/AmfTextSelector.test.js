import React from 'react'
import AmfTextSelector from './AmfTextSelector'
import renderer from 'react-test-renderer'

test('AmfTextSelector without validation', () => {
  let props = {
    label: 'Label',
    onPress: () => {},
    value: null,
		items: [
			{ label:'Option 1\nWith Enter\n', value:1 },
			{ label:'Option 2\nWith Enter\n', value:2 },
			{ label:'Option 3\nWith Enter\n', value:3 },
		]
	}

	const component = renderer.create(<AmfTextSelector {...props} />)

	let tree = component.toJSON()
	expect(tree).toMatchSnapshot()
})


