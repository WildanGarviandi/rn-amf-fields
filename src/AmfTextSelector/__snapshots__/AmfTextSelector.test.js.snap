// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AmfTextSelector without validation 1`] = `
<RCTScrollView
  contentContainerStyle={
    Object {
      "padding": 16,
    }
  }
  horizontal={true}
  style={undefined}
>
  <View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "marginHorizontal": 8,
          "opacity": 1,
          "width": 450,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        style={
          Array [
            Object {
              "borderTopLeftRadius": 4,
              "borderTopRightRadius": 4,
              "height": 10,
            },
            Object {
              "backgroundColor": "rgb(165, 0, 38)",
            },
          ]
        }
      />
      <View
        style={
          Array [
            Object {
              "borderBottomWidth": 1,
              "borderLeftWidth": 1,
              "borderRightWidth": 1,
              "padding": 16,
            },
            Object {
              "backgroundColor": "white",
              "borderColor": "#D4D4D4",
            },
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={false}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "black",
            }
          }
        >
          Option 1
          With Enter
          
        </Text>
      </View>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "marginHorizontal": 8,
          "opacity": 1,
          "width": 450,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        style={
          Array [
            Object {
              "borderTopLeftRadius": 4,
              "borderTopRightRadius": 4,
              "height": 10,
            },
            Object {
              "backgroundColor": "rgb(253, 190, 112)",
            },
          ]
        }
      />
      <View
        style={
          Array [
            Object {
              "borderBottomWidth": 1,
              "borderLeftWidth": 1,
              "borderRightWidth": 1,
              "padding": 16,
            },
            Object {
              "backgroundColor": "white",
              "borderColor": "#D4D4D4",
            },
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={false}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "black",
            }
          }
        >
          Option 2
          With Enter
          
        </Text>
      </View>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "marginHorizontal": 8,
          "opacity": 1,
          "width": 450,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        style={
          Array [
            Object {
              "borderTopLeftRadius": 4,
              "borderTopRightRadius": 4,
              "height": 10,
            },
            Object {
              "backgroundColor": "rgb(0, 104, 55)",
            },
          ]
        }
      />
      <View
        style={
          Array [
            Object {
              "borderBottomWidth": 1,
              "borderLeftWidth": 1,
              "borderRightWidth": 1,
              "padding": 16,
            },
            Object {
              "backgroundColor": "white",
              "borderColor": "#D4D4D4",
            },
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={false}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "black",
            }
          }
        >
          Option 3
          With Enter
          
        </Text>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
